algorithms/kernel/svm/oneapi/svm_train_cache_oneapi.h

```diff
@@ -0,0 +1,239 @@
+        return status;
+    }
+
+protected:
+    SVMCacheOneAPI(const size_t blockSize, const size_t lineSize, const NumericTablePtr & xTable, const kernel_function::KernelIfacePtr & kernel)
+        : super(blockSize, lineSize, kernel), _nSelectRows(0), _ifComputeSubKernel(false)
+    {}
+
+    services::Status init(const size_t cacheSize, const NumericTablePtr & xTable)
+    {
+        services::Status status;
+        auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+
+        _cache = context.allocate(TypeIds::id<algorithmFPType>(), _lineSize * _blockSize, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        _cacheBuff      = _cache.get<algorithmFPType>();
+        auto cacheTable = SyclHomogenNumericTable<algorithmFPType>::create(_cacheBuff, _lineSize, _blockSize, &status);
+
+        const size_t p = xTable->getNumberOfColumns();
+        _xBlock        = context.allocate(TypeIds::id<algorithmFPType>(), _blockSize * p, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        _xBlockBuff                    = _xBlock.get<algorithmFPType>();
+        const NumericTablePtr xWSTable = SyclHomogenNumericTable<algorithmFPType>::create(_xBlockBuff, p, _blockSize, &status);
+
+        DAAL_CHECK_STATUS_VAR(status);
+        _kernel->getParameter()->computationMode = kernel_function::matrixMatrix;
+        _kernel->getInput()->set(kernel_function::X, xWSTable);
+        _kernel->getInput()->set(kernel_function::Y, xTable);
+
+        kernel_function::ResultPtr shRes(new kernel_function::Result());
+        shRes->set(kernel_function::values, cacheTable);
+        _kernel->setResult(shRes);
+
+        return status;
+    }
+
+    services::Status initSubKernel(const size_t blockSize, const NumericTablePtr & xTable)
+    {
+        services::Status status;
+        auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+
+        auto cacheHalf = _cacheBuff.getSubBuffer(_lineSize * _nSelectRows, _lineSize * blockSize, &status);
+
+        auto cacheTable = SyclHomogenNumericTable<algorithmFPType>::create(cacheHalf, _lineSize, blockSize, &status);
+
+        const size_t p = xTable->getNumberOfColumns();
+        DAAL_CHECK_STATUS_VAR(status);
+
+        const NumericTablePtr xWSTable = SyclHomogenNumericTable<algorithmFPType>::create(_xBlockBuff, p, blockSize, &status);
+
+        DAAL_CHECK_STATUS_VAR(status);
+        _kernel->getParameter()->computationMode = kernel_function::matrixMatrix;
+        _kernel->getInput()->set(kernel_function::X, xWSTable);
+        _kernel->getInput()->set(kernel_function::Y, xTable);
+
+        kernel_function::ResultPtr shRes(new kernel_function::Result());
+        shRes->set(kernel_function::values, cacheTable);
+        _kernel->setResult(shRes);
+
+        return status;
+    }
+
+protected:
+    size_t _nSelectRows;
+    bool _ifComputeSubKernel;
+    UniversalBuffer _cache;
```

Comment:
Why do you need `_cache` in `noCache` case?

Reply:
what should be obtained by the result of the cache. In this case, at each external iteration, we compute a new one and return the result.

Reply:
It can be made as a cache. In the future, I think we will expand it. Now it is done for the future.